app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumCacheTest.kt

```diff
@@ -0,0 +1,166 @@
+        resetStorage()
+        setDatabase(PREFERENCES, MockDatabase())
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+
+    private fun resetOnlineForum() {
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+
+    @Test
+    fun newPostIsAddedToCache() {
+        // Add a post
+        val postOnline = cachedForum.newPost(AUTHOR, CONTENT, isPost = true).join()
+
+        // Reset the forum, acts as a clear of the data
+        resetOnlineForum()
+
+        // Fetch through cache the post
+        val postCached = cachedForum.getPost(postOnline.post.key).join()
+
+        assertEquals(CONTENT, postCached.post.content)
+        assertEquals(AUTHOR, postCached.post.author)
+    }
+
+    @Test
+    fun repliesAreAddedToCache() {
+        // Add a post
+        val postOnline = cachedForum.newPost(AUTHOR, CONTENT, isPost = true).join()
+
+        // Add two replies
+        postOnline.reply(AUTHOR, CONTENT).join()
+        postOnline.reply(AUTHOR, CONTENT).join()
+
+        // Reset the forum, acts as a clear of the data
+        resetOnlineForum()
+
+        // Fetch through cache the post
+        val postCached = cachedForum.getPost(postOnline.post.key).join()
+
+        assertEquals(2, postCached.replies.size)
+    }
+
+    @Test
+    fun postAndGetDoesNotDuplicate() {
+        // Add a post
+        val postOnline = cachedForum.newPost(AUTHOR, CONTENT, isPost = true).join()
+
+        // Fetch it again
+        cachedForum.getPost(postOnline.post.key).join()
+
+        // Reset the forum, acts as a clear of the data
+        resetOnlineForum()
+
+        // Fetch through cache the post
+        val all = cachedForum.getAll().join()
+
+        assertEquals(1, all.size)
+        assertEquals(1, all[0].second.size)
+    }
+
+    @Test
+    fun getAddsToCache() {
+        // Add an external post
+        val postOnline = rawForum.newPost(AUTHOR, CONTENT, isPost = true).join()
+
+        // Fetch it through the cache
+        cachedForum.getPost(postOnline.post.key).join()
+
+        // Reset the forum, acts as a clear of the data
+        resetOnlineForum()
+
+        // Fetch through cache the post
+        val cachedPost = cachedForum.getPost(postOnline.post.key).join()
+
+        assertEquals(CONTENT, cachedPost.post.content)
+        assertEquals(AUTHOR, cachedPost.post.author)
+    }
+
+    @Test
+    fun getAllAddsToCache() {
+        // Add an external post and reply
+        val postOnline = rawForum.newPost(AUTHOR, CONTENT, isPost = true).join()
+        postOnline.reply(AUTHOR, CONTENT).join()
+
+        // Fetch through cache the post
+        val all = cachedForum.getAll().join()
+
+        // Reset the forum, acts as a clear of the data
+        resetOnlineForum()

```

Comment:
shouldn't you call getAll again after resetting the online forum (you don't seem to be testing anything here since all takes value before the reset)

Reply:
Indeed, I switched the order of both lines.
Still confident the test will pass, since I spent a lot of time to make this work.

Reply:
I found a critical bug due to this. THANK YOU! :heart: